core/schema/container.go

```diff
@@ -1807,6 +1807,44 @@ func (s *containerSchema) asTarball(
 	parent dagql.Instance[*core.Container],
 	args containerAsTarballArgs,
 ) (inst dagql.Instance[*core.File], err error) {
+	if op := core.DagOpFromContext(ctx); op == nil {
+		fs, err := parent.Self.FSState()
+		if err != nil {
+			return inst, err
+		}
+
+		filename := "/container.tar"
+		st, err := core.DagLLB(ctx, s.srv,
+			[]dagql.Selector{
+				{
+					Field: "directory",
+				},
+				{
+					Field: "withFile",
+					Args: []dagql.NamedInput{
+						{
+							Name:  "path",
+							Value: dagql.String(filename),
+						},
+						{
+							Name:  "source",
+							Value: dagql.NewID[*core.File](dagql.CurrentID(ctx).WithMetadata("", true)),
```

Comment:
Ahaha, this is really neat :tada: I can do a self call here, forcing it to be tainted - the context it's re-evaluated in has a `DagOp`, so we then execute the other path.